nixos/modules/services/monitoring/vnstat.nix

```diff
@@ -0,0 +1,21 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.vnstat;
+in {
+  options.services.vnstat = {
+    enable = mkEnableOption "If enabled, vnstatd will be started to update the network usage statistics.";
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.vnstat = {
+      description = "vnStat network traffic monitor";
+      path = [ pkgs.coreutils ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig.ExecStart = "${pkgs.vnstat}/bin/vnstatd -n";
+      serviceConfig.ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
```

Comment:
systemd supports lots of mechanisms for isolating services. For example, you can do
```nix
let # for syntax highlighting only
serviceConfig = {
  ProtectHome = true;
  PrivateDevices = true;
  PrivateTmp = true;
}
```
to have the service run with an empty /home, a minimal /dev and a private /tmp. See `systemd.exec(5)` and `systemd.resource-control(5)` for details. As long as the service is running with full privileges, these directives mainly serve to reduce the risk of accidental data loss/information leaking, so it'd be better if it also ran under an unprivileged account.